src/components/hoc/LoadingIf.jsx

```diff
@@ -0,0 +1,67 @@
+"use strict";
+
+import React from 'react';
+
+const UPDATE_MS = 500;
+const MAX_DOTS = 3;
+
+var LoadingIf = (Wrapped, loadingCondition) => class LoadingIf extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            dots: '',
+            loading: loadingCondition(props)
+        }
+    }
+
+    componentDidMount() {
+        this.interval = setInterval(() => {
+            let currentDots = this.state.dots + '.';
```

Comment:
So maybe you could add support for a custom loading indicator and leave this as a default or smth.